src/pages/__test__/home.test.js

```diff
@@ -0,0 +1,305 @@
+import React from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import * as router from 'react-router';
+import HomePage from '../home';
+import * as BannersMock from '../../utils/hooks/useFeaturedBanners';
+import * as CategoriesMock from '../../utils/hooks/useFeaturedCategories';
+import * as ProductsMock from '../../utils/hooks/useProducts';
+import AppContext from '../../utils/appContext';
+
+describe('Home Page tests: ', () => {
+    const navigate = jest.fn();
+
+    beforeEach(() => {
+        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    test('Featured Banners Slider fetching and rendering data', async () => {
+        const mockDispatch = jest.fn();
+
+        const BannersSpy = jest.spyOn(BannersMock, 'default').mockReturnValue({
```

Comment:
You need to mock the return data from the API using `msw.js` or jest with the Axios Mock instead you are mocking the Hook.

Reply:
Apply this comment for all the tests.